ntoskrnl/lpc/complete.c

```diff
@@ -469,6 +547,27 @@ NtCompleteConnectPort(IN HANDLE PortHandle)
         return STATUS_SUCCESS;
     }
 
+    /* Ensure client thread is on the reply chain */
+    if (Port->ConnectionPort)
+    {
+        PLIST_ENTRY Entry;
+        for (Entry = Port->ConnectionPort->LpcReplyChainHead.Flink;
+             Entry != (PLIST_ENTRY)(&Port->ConnectionPort->LpcReplyChainHead.Flink);
```

Comment:
Casting here was added to fix compiler error about comparison of distinct pointer types.

Reply:
But there are not distinct, since LpcReplyChainHead is LIST_ENTRY and thus, &LpcReplyChainHead is a PLIST_ENTRY, and LpcReplyChainHead.Flink is already a PLIST_ENTRY.